homeassistant/components/touchlinesl/climate.py

```diff
@@ -0,0 +1,119 @@
+            for z in coordinators[module_id].data.zones
+        )
+
+
+CONST_TEMP_PRESET_NAME = "Constant Temperature"
+
+
+class TouchlineSLZone(CoordinatorEntity[TouchlineSLModuleCoordinator], ClimateEntity):
+    """Roth Touchline SL Zone."""
+
+    _attr_has_entity_name = True
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_mode = HVACMode.HEAT
+    _attr_hvac_modes = [HVACMode.HEAT]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+
+    def __init__(self, coordinator: TouchlineSLModuleCoordinator, zone_id: int) -> None:
+        """Construct a Touchline SL climate zone."""
+        super().__init__(coordinator, context=zone_id)
+        self.zone_id: int = zone_id
+
+        self._attr_name = None
+        self._attr_unique_id = f"{self.zone_id}"
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(coordinator.data.zones[zone_id].id))},
+            name=coordinator.data.zones[zone_id].name,
+            manufacturer="Roth",
+            via_device=(DOMAIN, coordinator.data.module.id),
+            model="zone",
+        )
+
+        # Call this in __init__ so data is populated right away, since it's
+        # already available in the coordinator data.
+        self.set_attr()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self.set_attr()
+        super()._handle_coordinator_update()
+
+    @property
+    def available(self) -> bool:
+        """Return if the device is available."""
+        return super().available and self.zone_id in self.coordinator.data.zones
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
+            return
+
+        await self._zone.set_temperature(temperature)
+        await self.coordinator.async_request_refresh()
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Assign the zone to a particular global schedule."""
+        if not self._zone:
+            return
+
+        if preset_mode == CONST_TEMP_PRESET_NAME and self._attr_target_temperature:
+            await self._zone.set_temperature(temperature=self._attr_target_temperature)
+            await self.coordinator.async_request_refresh()
+            return
+
+        if schedule := self.coordinator.data.schedules[preset_mode]:
+            await self._zone.set_schedule(schedule_id=schedule.id)
+            await self.coordinator.async_request_refresh()
+
+    def set_attr(self) -> None:
+        """Populate attributes with data from the coordinator."""
+        zone: Zone = self.coordinator.data.zones[self.zone_id]
+        schedule_names = self.coordinator.data.schedules.keys()
+
+        self._zone = zone
```

Comment:
instead of setting zone, create a property like the `device` property in `fujitsu_fglair`. It allows you to just access `self.zone` everywhere and it will get it from the coordiantor